packages/custom_lint/lib/src/v2/server_to_client_channel.dart

```diff
@@ -222,15 +222,7 @@ void _writePackageConfigForTempProject(
       join(contextRoot.root, 'pubspec.yaml'),
     );
 
-    String packageConfigContent;
-    try {
-      packageConfigContent = packageConfigFile.readAsStringSync();
-    } on FileSystemException {
-      throw StateError(
-        'No package_config.json found. Did you forget to run `pub get`?\n'
-        'Tried to look in:\n${contextRoots.map((e) => '- ${e.root}\n').join()}',
-      );
-    }
```

Comment:
This check is already done in `getPackageListForContextRoots`, it can never fail here.